### PR TITLE
Updated to KafkaProducer Java API and minor improvement to Simple Consumer

### PR DESCRIPTION
**Pull request by Alex-Cook4**
New Java API which improves performance ~50x. 
Improved Simple Consumer performance slightly and now support writing topic out as an attribute.

### CODE REVIEW

File: com.ibm.streamsx.messaging/impl/java/src/com/ibm/streamsx/messaging/kafka/KafkaClient.java [hunk 1]
```diff
@@ -184,96 +166,50 @@ public void run() {
...
-		}
-		List<KeyedMessage<String, String> > lst = 
-				new ArrayList<KeyedMessage<String,String>>(topics.size());
-		for(String topic : topics) {
-			lst.add(new KeyedMessage<String, String>(topic, key, data));
+	void send(Tuple tuple, List<String> topics) throws Exception {
+		//repeated code for performance reasons
+		if(keyAH.isAvailable()) {
+			for(String topic : topics) {
+				producer.send(new ProducerRecord(topic,keyAH.getBytes(tuple), messageAH.getBytes(tuple)));
```
Comment (dlaboss) [inline com.ibm.streamsx.messaging/impl/java/src/com/ibm/streamsx/messaging/kafka/KafkaClient.java]: The key and msg bytes are static during the loop.  If send() doesn't modify them (why would it?), getting them once outside the loop will be better.

**Comment by Alex-Cook4** [comment]
Good catch Dale. Thanks.

**Comment by dlaboss** [comment]
Do the new Kafka API/jars still support the old config properties, or does the messaging tk provide backward compat support for them,  or will existing clients of the tk stop working with this new version and have to be edited / rebuilt to use the new properties?

**Comment by Alex-Cook4** [comment]
Old properties will not be supported. There is no backward compatibility. Existing clients will need to change config file to these properties: http://kafka.apache.org/documentation.html#newproducerconfigs

**Comment by dlaboss** [comment]
Are you sure the tk is allowed to violate backwards compat here?

Based on a quick review of the new producer info, it looks like the tk could perform a small amount of property mapping and then continue to support existing clients -- at least those that only use common properties.

E.g.,
* the old required `metadata.broker.list` looks like it's simply rename to `bootstrap.servers`
* `serializer.class` is gone (and ignored? or easily filtered out)
* `request.required.acks` is renamed to `acks`.  The default has changed from `0` to `1`, and the old `-1` value seems to now be represented as `all`.

**Comment by Alex-Cook4** [comment]
We do not plan to support properties for this reason: 
1. Mapping the two main required properties wouldn't be too hard, but then customers could be blinded to the fact that all of their other configurations are getting ignored.
2. Not providing any backwards compatibility ensures that customers will update to the new configurations for all important aspects.
3. Mapping all old properties to new ones would be challenging because there are 18 old configs that don't all directly correlate to new configs (even values have to be mapped differently in some cases)

**Pull request merged**